### DISCUSSION TITLE
自荐一个扩展 Skynet 的方案

### DISCUSSION THREAD

**Discussion by Zwlin98** [cloudwu/skynet#1905] [Ideas]
[Moon](https://github.com/Zwlin98/moon) 是用 Go 写的一个 Skynet 的辅助程序，可以加入 Skynet 的 Cluster 集群，支持 Skynet 的 RPC 语法与 Skynet 集群互相调用。可以替代 Skynet（C + Lua）实现较为繁琐的一些辅助逻辑，如向外发起 https/http 请求，集成第三方 SDK 等。

**Comment by hanxi** [comment]
我也写了一个玩具，ts 语言的。lua-seri 是直接让 ai 照着你的代码翻译的。 https://github.com/hanxi/bun-skynet

**Reply by najoast** [reply]
skynet 的 TypeScript 方案已经有人做了：https://github.com/lsg2020/skynet_ts

**Reply by hanxi** [reply]
发现早就 star 过了，看了一下没有实现 cluster 。我这个只是一个玩具，纯 ts 写的，就是为了玩玩前端的新玩具 bun 。

**Comment by sniper00** [comment]
扩展生态方面，我认为rust写lua扩展库的方式更适合
https://github.com/sniper00/moon/wiki/Writing-Lua-extension-library-using-Rust
https://github.com/sniper00/moon-extensions

**Comment by Zwlin98** [comment]
思路还是不太一样，我这个一开始只是为了更方便的集成一些三方 SDK，完全不侵入 skynet。仅仅是用 go 实现了一套 skynet 的 cluster 协议，可以加入 skynet 的集群之中，对运行在 skynet 里的 lua 服务而言，表现的和其他 skynet 节点上的 lua 服务没有区别。

**Comment by 253980289** [comment]
skynet本身也提供了http相关基础功能模块，并且也支持https，如果缺少一些通用加解密算法的话，那么可以考虑加入第三方openss的lua绑定模块，我们目前就是这么用的，毕竟一套语言和服务维护相对简便。
